flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/ExecutionDecorator.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.runtime.tasks;
+
+import org.apache.flink.streaming.runtime.tasks.mailbox.Mail;
+import org.apache.flink.util.function.ThrowingRunnable;
+
+import java.util.concurrent.Callable;
+
+/**
+ * Wraps execution of a {@link Runnable}, {@link ThrowingRunnable}, {@link Callable}, or {@link Mail}.
+ * Intended to customize execution in sub-types fo {@link org.apache.flink.streaming.runtime.tasks.StreamTask StreamTask},
+ * e.g. synchronization in {@link org.apache.flink.streaming.runtime.tasks.SourceStreamTask SourceStreamTask}.
+ */
+public interface ExecutionDecorator {
+	void run(Runnable runnable);
+
+	<E extends Throwable> void runThrowing(ThrowingRunnable<E> runnable) throws E;
+
+	<R> R call(Callable<R> callable) throws Exception;
+
+	void dispatch(Mail mail);
+
+	ExecutionDecorator NOP = new ExecutionDecorator() {
```

Comment:
NOP = no operation? that's some mock implementation that does not invoke anything for me.
Wouldn't it be UNSYNCHRONIZED?

Reply:
Yes, NOP is no operation. Unsynchronized sounds too specific to me.

Reply:
But NOP is really wrong as it is performing these actions.
`public void run(Runnable runnable) {}` is a nop executor for me.